fdbrecord/src/main/java/com/yahoo/ycsb/db/fdbrecord/FDBRecordClient.java

```diff
@@ -0,0 +1,309 @@
+          .setPrimaryKey(Key.Expressions.field("YCSB"));
+
+      recordStoreProvider = context -> FDBRecordStore.newBuilder()
+          .setMetaDataProvider(metaDataBuilder.getRecordMetaData())
+          .setContext(context)
+          .setKeySpacePath(keySpacePath)
+          .createOrOpen();
+
+    } catch (RecordCoreException e) {
+      logger.error(MessageFormatter.format("Error in database operation: {}", "init").getMessage(), e);
+      throw new DBException(e);
+    } catch (Descriptors.DescriptorValidationException e) {
+      throw new DBException(e);
+    }
+  }
+
+  @Override
+  public void cleanup() throws DBException {
+    if (batchCount > 0) {
+      batchInsert();
+      batchCount = 0;
+    }
+    try {
+      db.close();
+    } catch (RecordCoreException e) {
+      logger.error(MessageFormatter.format("Error in database operation: {}", "cleanup").getMessage(), e);
+      throw new DBException(e);
+    }
+  }
+
+  private Message getRecord(String key, Map<String, ByteIterator> values) {
+    msgBuilder.clear();
+    msgBuilder.setField(msgDesc.findFieldByName("YCSB"), key);
+    for (Map.Entry<String, ByteIterator> ent : values.entrySet()) {
+      msgBuilder.setField(msgDesc.findFieldByName(ent.getKey()), ent.getValue().toString());
+    }
+    return msgBuilder.build();
+  }
+
+  private Status convRecordToMap(Message record, Set<String> fields, Map<String, ByteIterator> result) {
+    for (Map.Entry<Descriptors.FieldDescriptor, Object> ent : record.getAllFields().entrySet()) {
+      String field = ent.getKey().getName();
+      String value = (String) ent.getValue();
+      result.put(field, new StringByteIterator(value));
+    }
+    if (fields != null) {
+      for (String field : fields) {
+        if (result.get(field) == null) {
+          logger.debug("field not found: {}", field);
+          return Status.NOT_FOUND;
+        }
+      }
+    }
+    return Status.OK;
+  }
+
+  private void batchInsert() {
+    try {
+      db.run(context -> {
+          FDBRecordStore recordStore = recordStoreProvider.apply(context);
+
+          for (int i = 0; i < batchCount; ++i) {
+            recordStore.insertRecord(batchRecords.get(i));
+          }
+
+          return null;
+        });
+    } catch (Throwable e) {
+      for (int i = 0; i < batchCount; ++i) {
+        logger.error(MessageFormatter.format("Error batch inserting key {}", batchRecords.get(i)).getMessage(), e);
+      }
+    } finally {
+      batchRecords.clear();
+    }
+  }
+
+  @Override
+  public Status insert(String table, String key, Map<String, ByteIterator> values) {
+    Message record = getRecord(key, values);
+    String rowKey = (String) record.getField(msgDesc.findFieldByName("YCSB"));
+    logger.debug("insert key = {}", rowKey);
+    try {
+      batchRecords.add(record);
+      batchCount++;
+      if (batchSize == 0 || batchSize == batchCount) {
+        batchInsert();
+        batchCount = 0;
+      }
+      return Status.OK;
```

Comment:
When batched but not executed this should be BATCHED_OK.

Reply:
Actually I copy this logic from `foundationdb-binding` driver. But I think it doesn't fully follow the interface. Which means this is not the batch behavior defined in the interface, but a kind of new batch behavior.
In this case, I cannot change it to `BATCHED_OK`, it will ruin the normal inserting behavior.
Or maybe I wrongly understand the interface?